### PR TITLE
Detailed jupyter error log

### PR DESCRIPTION
**Pull request by SmartManoj**

### CODE REVIEW

File: opendevin/runtime/plugins/jupyter/execute_cli.py [hunk 1]
```diff
@@ -14,21 +15,26 @@ def execute_code(code, print_output=True):
 
     # Set the default kernel ID
     kernel_id = 'default'
-
-    for i in range(10):
+    output = ''
+    for i in range(3):
```
Comment (xingyaoww) [inline opendevin/runtime/plugins/jupyter/execute_cli.py:19]: Any particular reason to set this to 3?

File: opendevin/runtime/plugins/jupyter/execute_cli.py [hunk 2]
```diff
@@ -14,21 +15,26 @@ def execute_code(code, print_output=True):
 
     # Set the default kernel ID
     kernel_id = 'default'
-
-    for i in range(10):
+    output = ''
```
Comment (xingyaoww) [inline opendevin/runtime/plugins/jupyter/execute_cli.py:18]: Is this line necessary?

File: opendevin/runtime/plugins/jupyter/execute_cli.py [hunk 3]
```diff
@@ -14,21 +15,26 @@ def execute_code(code, print_output=True):
...
-            pass
+            if i == 2:
+                traceback.print_exc()
         time.sleep(2)
     else:
-        print('Failed to connect to the Jupyter server')
+        if not output:
+            with open('/opendevin/logs/jupyter_execute_server.log', 'r') as f:
+                output = f.read()
+        print('Failed to connect to the Jupyter server', output)
```
Comment (xingyaoww) [inline opendevin/runtime/plugins/jupyter/execute_cli.py:37]: Ahh this will help a ton in the future for jupyter debugging

**Review by xingyaoww — COMMENTED** [review]
Mostly LGTM

File: opendevin/runtime/plugins/jupyter/execute_cli.py (hunk 1 above)
Comment (SmartManoj) [reply opendevin/runtime/plugins/jupyter/execute_cli.py:19]: Previously I increased from 5 to 10.  Reason for the 5?

File: opendevin/runtime/plugins/jupyter/execute_cli.py (hunk 2 above)
Comment (SmartManoj) [reply opendevin/runtime/plugins/jupyter/execute_cli.py:18]: Yes, output won't be defined if connection problem. Better to move into except block

File: opendevin/runtime/plugins/jupyter/execute_cli.py (hunk 3 above)
Comment (tobitege) [reply opendevin/runtime/plugins/jupyter/execute_cli.py:37]: Is it guaranteed that that server.log file exists? Just wondering.

**Review by neubig — APPROVED** [review]
LGTM!

**Pull request merged**